packages/insomnia-app/app/models/helpers/settings.ts

```
@@ -0,0 +1,213 @@
+import { readFileSync } from 'fs';
+import { Settings } from 'insomnia-common';
+import { InsomniaConfig, validate } from 'insomnia-config';
+import { resolve } from 'path';
+import { mapObjIndexed, mergeRight, once } from 'ramda';
+import { omitBy } from 'ramda-adjunct';
+
+import { isDevelopment } from '../../common/constants';
+import { getDataDirectory, getPortableExecutableDir } from '../../common/electron-helpers';
+
+/** takes an unresolved (or resolved will work fine too) filePath of the insomnia config and reads the insomniaConfig from disk */
+const readConfigFile = (filePath: string) => {
+  try {
+    const resolvedFilePath = resolve(filePath);
+    const fileContents = readFileSync(resolvedFilePath, 'utf-8');
+    return JSON.parse(fileContents) as unknown;
+  } catch (error: unknown) {
+    return undefined;
```

Should probably `console.error` here? If the JSON is malformed (for example), we swallow the error entirely.

---

Thanks, this is now covered by INS-1046 (along with invalid configs that are valid json, covered by INS-1030).  As a stop-gap until INS-1046, I think what you're asking for is reasonable, and, after all, we already do something similar in `getConfigSettings` for configs that fail validation.

see: c15656c6bce18a65267f1251e859c89fb74ab3de